src/components/ProviderInitializer/ProviderInitializer.tsx

```diff
@@ -136,6 +145,22 @@ export function ProviderInitializer() {
     }
   }, [isLoggedIn, userAccount]);
 
+  useEffect(() => {
+    const shouldGetGasMetadata =
+      isLoggedIn && userAccount.shard != null && !network.gasStationMetadata;
+
+    if (!shouldGetGasMetadata) {
+      return;
+    }
+
+    updateGasStationMetadata();
+  }, [
+    userAccount.shard,
+    userAccount.address,
+    isLoggedIn,
+    network.gasStationMetadata
```

Comment:
I think this needs to be removed from deps

Reply:
1. user loggs in
2. gas is undefined
3. gas is feched
4. gas is changed
5. undefined != changed
6. effect is triggred again